File: solutions/typescript/allergies/1/allergies.ts
enum Allergen {
  Eggs = "eggs",
  Peanuts = "peanuts",
  Shellfish = "shellfish",
  Strawberries = "strawberries",
  Tomatoes = "tomatoes",
  Chocolate = "chocolate",
  Pollen = "pollen",
  Cats = "cats"
}

const allergenToBitMask: Record<Allergen,number> = {
  [Allergen.Eggs]:           0b00000001,
  [Allergen.Peanuts]:        0b00000010,
  [Allergen.Shellfish]:      0b00000100,
  [Allergen.Strawberries]:   0b00001000,
  [Allergen.Tomatoes]:       0b00010000,
  [Allergen.Chocolate]:      0b00100000,
  [Allergen.Pollen]:         0b01000000,
  [Allergen.Cats]:           0b10000000,
}

export class Allergies {
  private allergenIndex: number = 0;
  
  constructor(allergenIndex: number) {
    this.allergenIndex = allergenIndex;
  }

  public list(): string[] {
    return Object.keys(allergenToBitMask)
      .filter(allergen => (this.allergenIndex & allergenToBitMask[allergen as Allergen]) === allergenToBitMask[allergen as Allergen])
  }

  public allergicTo(allergen: Allergen): boolean {
    return (this.allergenIndex & allergenToBitMask[allergen]) === allergenToBitMask[allergen];
  }
}

File: solutions/python/allergies/1/allergies.py
class Allergies:
    allergen_masks = {
        "eggs": 0b0000_0001,
        "peanuts": 0b0000_0010,
        "shellfish": 0b0000_0100,
        "strawberries": 0b0000_1000,
        "tomatoes": 0b0001_0000,
        "chocolate": 0b0010_0000,
        "pollen": 0b0100_0000,
        "cats": 0b1000_0000,
    }

    def __init__(self, score: int):
        self.score = score & 0xFF  # Only consider the lowest 8 bits
        self.lst: list[str] = [
            allergen
            for allergen, mask in self.allergen_masks.items()
            if self.score & mask
        ]

    def allergic_to(self, item: str) -> bool:
        return item in self.lst
